internal/services/kusto/kusto_cluster_resource.go

```diff
@@ -238,6 +238,13 @@ func resourceKustoCluster() *pluginsdk.Resource {
 			"tags": tags.Schema(),
 		},
 	}
+
+	if features.FourPointOhBeta() {
+		s.Schema["engine"].Default = string(kusto.EngineTypeV3)
+	}
+	s.Schema["engine"].Default = string(kusto.EngineTypeV2)
```

Comment:
thank you @katbyte  for pushing this fix.
question : should there be an `else` clause here? so that we don't always set the Default to `EngineTypeV2`, rather only if `features.FourPointOhBeta()` is false?

Reply:
there sure should be! thanks :)